src/Container/Container.php

```diff
@@ -36,76 +41,128 @@ class Container implements ContainerInterface
      * @param mixed  $value
      * @return Container
      */
-    public function set($key, $value)
+    public function set(string $key, $value): self
     {
+        if (!(\is_string($value) || \is_object($value))) {
+            throw new \InvalidArgumentException('Value must be a class name, an object instance, or a callable');
+        }
+
         $this->dicValues[$key] = $value;
         unset($this->dicObjects[$key]); // In case an object instance was stored for sharing
 
         return $this;
     }
 
+    public function alias(string $alias, string $key): self {
+        $this->aliases[$alias] = $key;
+
+        return $this;
+    }
+
     /**
      * Check if a DIC value/object exists.
      *
      * @param string $key
      * @return bool
      */
-    public function has($key)
+    public function has($key): bool
     {
-        return array_key_exists($key, $this->dicValues);
+        if (array_key_exists($key, $this->aliases)) {
+            $key = $this->aliases[$key];
+        }
+
+        if (array_key_exists($key, $this->dicValues)) {
+            return true;
+        }
+
+        return class_exists($key);
```

Comment:
what is the impact of this return statement? The container will return true if we pass a FQCN to it which is not added to the container right? Can this have negative consequences? Feels like cheating ;-)

Reply:
You mat be right about that. Only returning true for defined keys would be more useful.